CircleSpawn/CircleSpawn/CircleSpawnController.swift

```diff
@@ -1,11 +1,128 @@
 import UIKit
 
-class CircleSpawnController: UIViewController {
-
-	// TODO: Assignment 1
+class CircleSpawnController: UIViewController, UIGestureRecognizerDelegate {
 
 	override func loadView() {
 		view = UIView()
 		view.backgroundColor = .white
 	}
+    
+    override func viewDidLoad() {
+        let doubleTap = UITapGestureRecognizer(target: self, action: #selector(handleDoubleTap(sender:)))
+        doubleTap.numberOfTapsRequired = 2
+        doubleTap.name = "double"
+        doubleTap.delegate = self
+        view.addGestureRecognizer(doubleTap)
+    }
+    
+    @objc func handleDoubleTap(sender: UITapGestureRecognizer) {
+        let location = sender.location(in: view)
+        let newCircle = UIView(frame: CGRect(origin: .zero, size: CGSize(width: 100, height: 100)))
+        newCircle.center = location
+        newCircle.layer.cornerRadius = 50
+        newCircle.backgroundColor = UIColor.randomBrightColor()
+        newCircle.alpha = 0
+        newCircle.transform = CGAffineTransform(scaleX: 2.0, y: 2.0)
+        newCircle.isUserInteractionEnabled = true
+        view.addSubview(newCircle)
+        
+        UIView.animate(withDuration: 0.15, animations: {
+            newCircle.alpha = 1
+            newCircle.transform = CGAffineTransform(scaleX: 1, y: 1)
+        })
+        
+        let tripleTap = UITapGestureRecognizer(target: self, action: #selector(handleTripleTap(sender:)))
+        tripleTap.numberOfTapsRequired = 3
+        tripleTap.name = "triple"
+        tripleTap.delegate = self
+        newCircle.addGestureRecognizer(tripleTap)
+        
+        let press = UILongPressGestureRecognizer(target: self, action: #selector(handlePress(sender:)))
+        press.minimumPressDuration = 0.15
+        press.delegate = self
+        newCircle.addGestureRecognizer(press)
+        
+        let pan = UIPanGestureRecognizer(target: self, action: #selector(handlePan(sender:)))
+        pan.delegate = self
+        newCircle.addGestureRecognizer(pan)
```

Comment:
Pan Gesture Recognizer zupełnie nie był tutaj potrzebny. Przez to zachowanie aplikacji jest trochę inne niż zakładane.

Przenoszenie należało rozpoznawać **tylko** przy pomocy `UILongPressGestureRecognizer`.